test/e2e/packagemanifest_e2e_test.go

```diff
@@ -3,170 +3,181 @@ package e2e
 			}
-			return false, nil
 		})
-		require.NoError(GinkgoT(), err, "package-server not available")
-
-		_, cleanupCatalogSource := createInternalCatalogSource(GinkgoT(), c, crc, catalogSourceName, testNamespace, manifests, []apiextensions.CustomResourceDefinition{crd}, []v1alpha1.ClusterServiceVersion{csv})
-		require.NoError(GinkgoT(), err)
-		defer cleanupCatalogSource()
-
-		_, err = fetchCatalogSource(GinkgoT(), crc, catalogSourceName, testNamespace, catalogSourceRegistryPodSynced)
-		require.NoError(GinkgoT(), err)
-
-		pm, err := fetchPackageManifest(GinkgoT(), pmc, testNamespace, packageName, packageManifestHasStatus)
-		require.NoError(GinkgoT(), err, "error getting package manifest")
-		require.NotNil(GinkgoT(), pm)
-		require.Equal(GinkgoT(), packageName, pm.GetName())
-		require.Equal(GinkgoT(), expectedStatus, pm.Status)
-		require.Equal(GinkgoT(), "0.0.0", pm.Status.Channels[0].CurrentCSVDesc.MinKubeVersion)
-		require.Equal(GinkgoT(), *dummyImage, pm.Status.Channels[0].CurrentCSVDesc.RelatedImages[0])
-		require.Equal(GinkgoT(), csv.Spec.NativeAPIs, pm.Status.Channels[0].CurrentCSVDesc.NativeAPIs)
-		require.Equal(GinkgoT(), "label", pm.GetLabels()["projected"])
-		require.Equal(GinkgoT(), "supported", pm.GetLabels()["operatorframework.io/arch.amd64"])
-		require.Equal(GinkgoT(), "supported", pm.GetLabels()["operatorframework.io/os.linux"])
-
-		// Get a PackageManifestList and ensure it has the correct items
-		pmList, err := pmc.OperatorsV1().PackageManifests(testNamespace).List(context.TODO(), metav1.ListOptions{})
-		require.NoError(GinkgoT(), err, "could not access package manifests list meta")
-		require.NotNil(GinkgoT(), pmList.ListMeta, "package manifest list metadata empty")
-		require.NotNil(GinkgoT(), pmList.Items)
-	})
-	It("loading relatedImages", func() {
-
-		defer cleaner.NotifyTestComplete(GinkgoT(), true)
-
-		sourceName := genName("catalog-")
-		packageName := "etcd-test"
-		image := "quay.io/olmtest/catsrc-update-test:related"
-
-		crc := newCRClient(GinkgoT())
-		pmc := newPMClient(GinkgoT())
-
-		source := &v1alpha1.CatalogSource{
-			TypeMeta: metav1.TypeMeta{
-				Kind:       v1alpha1.CatalogSourceKind,
-				APIVersion: v1alpha1.CatalogSourceCRDAPIVersion,
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      sourceName,
-				Namespace: testNamespace,
-				Labels:    map[string]string{"olm.catalogSource": sourceName},
-			},
-			Spec: v1alpha1.CatalogSourceSpec{
-				SourceType: v1alpha1.SourceTypeGrpc,
-				Image:      image,
-			},
-		}
 
-		expectedRelatedImages := map[string]string{
-			"quay.io/coreos/etcd@sha256:3816b6daf9b66d6ced6f0f966314e2d4f894982c6b1493061502f8c2bf86ac84":          "",
-			"quay.io/coreos/etcd@sha256:49d3d4a81e0d030d3f689e7167f23e120abf955f7d08dbedf3ea246485acee9f":          "",
-			"quay.io/coreos/etcd-operator@sha256:c0301e4686c3ed4206e370b42de5a3bd2229b9fb4906cf85f3f30650424abec2": "",
-		}
+		It("lists the CatalogSource contents using the PackageManifest API", func() {
```

Comment:
Is this two specs in one ("packagemanifest can be retrieved by name" and "packagemanifest appears in list")? If yes, I think the second one is underspecified -- it would pass if `Items` is an empty slice or if it contained a packagemanifest other than the expected one.

Reply:
I split them into two `It` specs